[Draft] Clarify serialization and discoverability (#40)
frictionlessdata/datapackage: content/docs/specifications/data-package.md

### Descriptor

On logical level, Data Package descriptor is represented by a data structure. The data structure `MUST` be a JSON-serializable `object` as defined in [RFC 4627](http://www.ietf.org/rfc/rfc4627.txt).

On physical level, Data Package descriptor is represented by a file. A data producer `MAY` use any suitable serialization format and `SHOULD` name the file `datapackage.json`. A data consumer `MUST` support JSON serialization format and `MAY` support other serialization formats like YAML or TOML.

> The above states that JSON is the only serialization format that `MUST` be used for publishing a Data Package while other serialization formats can be used in projects or systems internally if supported by corresponding implementations.

Reviewer: I think this is a good requirement, but imo it doesn't follow from what is stated above. Can we shorten to:

```suggestion
JSON is the serialization format that `MUST` be used for publishing a Data Package while other serialization formats can be used in projects or systems internally if supported by corresponding implementations.
```

Reviewer: The same correction should be applied to the other specifications.

> This specification does not define any discoverability mechanisms making a serialized Data Package be referenced only directly by its URI. It means that techically the name of a Data Package file is irrelevant although it is good practice to use `datapackage.json` as a public convention.

Reviewer: The first sentence is somewhat confusing to me. A suggestion (that also removes the typo "techically"):

```suggestion
This specification does not define any discoverability mechanisms. Any URI can be used to directly reference a serialized Data Package. It is good practice and a common convention to name the file `datapackage.json`.
```
